Add ssg.products.get_all to return linux_os, other products

cipherboy opened the pull request:
This function has been helpful in both #3178 and #3193, so I've decided to split it off as a separate commit.

In essence, we wish to figure out which products belong to the `linux_os` shared guide and which have their own guides. Since this is an expensive operation (and changes based on the contents of the `product.yml`), it makes sense to have this be a function vs either storing the results as constants, or having these be static constants.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/94fcec2a-84ae-44f3-81da-6d2973b9914e) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/94fcec2a-84ae-44f3-81da-6d2973b9914e) completed: **1 updated code elements**

cipherboy commented:
`hostname: dd688bc3-fe57-4b36-a47b-b4079815db0f@1.production-1-worker-org-a-6-gce` CentOS failure:

```python
[  0%] [firefox-content] generating oval-unlinked.xml
Traceback (most recent call last):
  File "/home/oscap/scap-security-guide/build-scripts/combine_ovals.py", line 100, in <module>
    main()
  File "/home/oscap/scap-security-guide/build-scripts/combine_ovals.py", line 56, in main
    args.ovaldirs)
  File "/home/oscap/scap-security-guide/ssg/build_ovals.py", line 262, in checks
    os.path.join(oval_dir, filename), env_yaml
  File "/home/oscap/scap-security-guide/ssg/jinja.py", line 89, in process_file
    template = _get_jinja_environment(substitutions_dict).get_template(filepath)
  File "/usr/lib64/python2.6/site-packages/jinja2/environment.py", line 552, in get_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/usr/lib64/python2.6/site-packages/jinja2/loaders.py", line 102, in load
    bucket = bcc.get_bucket(environment, name, filename, source)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 166, in get_bucket
    self.load_bytecode(bucket)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 234, in load_bytecode
    bucket.load_bytecode(f)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 71, in load_bytecode
    self.code = marshal.load(f)
EOFError: EOF read where object expected
```

Restarting.

matejak commented on ssg/products.py:
```diff
@@ -40,6 +42,36 @@ def parse_name(product):
+    those which use linux_os and those which use their own directory. Returns
+    a tuple of sets, (linux_products, other_products).
+    """
+    directories = ['debian8', 'fedora', 'ol7', 'opensuse', 'rhel6',
```
This is a useful collection, what about putting this to the constants module?

matejak commented on ssg/products.py:
```diff
@@ -40,6 +42,36 @@ def parse_name(product):
+    """
+    Analyzes all products in the SSG root and sorts them into two categories:
+    those which use linux_os and those which use their own directory. Returns
+    a tuple of sets, (linux_products, other_products).
```
I find a tuple quite restrictive, what about returning a name tuple or dictionary? Then, if the need arises, one can introduce `all`, `browsers` or `java` sets :-)

pep8speaks commented:
Hello @cipherboy! Thanks for updating the PR.

 - In the file [`ssg/constants.py`](https://github.com/OpenSCAP/scap-security-guide/blob/160ef214e874993915ed75d05aa4d6bd2a77c6fd/ssg/constants.py), following are the PEP8 issues :

> [Line 138:100](https://github.com/OpenSCAP/scap-security-guide/blob/160ef214e874993915ed75d05aa4d6bd2a77c6fd/ssg/constants.py#L138): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 99 characters)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/68be8146-ebf4-45b9-ade0-b2fb2c34ff4e) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/2efdbaae-3db5-46a4-8633-e966ee030b28) was created.

cipherboy commented:
@matejak Okay! Thanks for the feedback :)

I've made it a `namedtuple` and updated the documentation. I moved the constant into `ssg.constants`, and I added a unit test to verify assertions about `fedora`, `rhel7`, and `jre`. :)

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/68be8146-ebf4-45b9-ade0-b2fb2c34ff4e) completed: **1 new issues, 3 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/OpenSCAP/scap-security-guide/inspections/2efdbaae-3db5-46a4-8633-e966ee030b28) completed: **1 new issues, 3 updated code elements**

matejak commented:
Great, I like those changes very much and I am merging this. The Travis CentOS CI failure is unrelated.

cipherboy commented:
Agreed:
`hostname: 72d8e875-1489-4469-8ac3-6412fb3c3c70@1.production-4-worker-org-a-3-gce` 

```python
Traceback (most recent call last):
  File "/home/oscap/scap-security-guide/build-scripts/combine_ovals.py", line 100, in <module>
    main()
  File "/home/oscap/scap-security-guide/build-scripts/combine_ovals.py", line 56, in main
    args.ovaldirs)
  File "/home/oscap/scap-security-guide/ssg/build_ovals.py", line 262, in checks
    os.path.join(oval_dir, filename), env_yaml
  File "/home/oscap/scap-security-guide/ssg/jinja.py", line 89, in process_file
    template = _get_jinja_environment(substitutions_dict).get_template(filepath)
  File "/usr/lib64/python2.6/site-packages/jinja2/environment.py", line 552, in get_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "/usr/lib64/python2.6/site-packages/jinja2/loaders.py", line 102, in load
    bucket = bcc.get_bucket(environment, name, filename, source)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 166, in get_bucket
    self.load_bytecode(bucket)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 234, in load_bytecode
    bucket.load_bytecode(f)
  File "/usr/lib64/python2.6/site-packages/jinja2/bccache.py", line 71, in load_bytecode
    self.code = marshal.load(f)
EOFError: EOF read where object expected
```

Restarting.
